src/gui/DatabaseTabWidget.cpp

```diff
@@ -829,6 +830,17 @@ void DatabaseTabWidget::updateLastDatabases(const QString& filename)
     }
 }
 
+void DatabaseTabWidget::updateLastDatabases(int index) {
+    if (index == -1) {
```

Comment:
You'll want to check that the index is out of range entirely, so index < 0 || index >= count()

You also need to check that you have a count at all, if there are no database widgets you should just return. 

You can also do this by trying to get the database widget by index, of nullptr then try `currentDatabaseWidget` if nullptr return, otherwise check for a file name and if present continue with the function.

Reply:
Thank you for ideas, I will try...

Reply:
Check my latest update, I found an example without `index` and as for me it looks more concise. Let me know if a variant with `index` is better.
Also I found a reason why it did not work yesterday: in the `Database` class I changed the order of `setFilePath()` and `markAsClean()`. Now it works.

Reply:
That is a good catch with the mark as clean!